mobile/src/main/java/rs/readahead/washington/mobile/views/settings/HideBehindCalculator.kt

```diff
@@ -4,70 +4,43 @@ import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
-import androidx.core.content.ContextCompat
-import com.hzontal.tella_locking_ui.CALCULATOR_ALIAS
-import org.hzontal.shared_ui.bottomsheet.BottomSheetUtils
-import rs.readahead.washington.mobile.MyApplication
+import androidx.core.os.bundleOf
 import rs.readahead.washington.mobile.R
-import rs.readahead.washington.mobile.bus.event.CamouflageAliasChangedEvent
-import rs.readahead.washington.mobile.util.CamouflageManager
+import rs.readahead.washington.mobile.databinding.OnboardCalculatorFragmentBinding
 import rs.readahead.washington.mobile.views.base_ui.BaseFragment
+import kotlin.properties.Delegates
 
 class HideBehindCalculator : BaseFragment() {
 
-    private val cm = CamouflageManager.getInstance()
+    private lateinit var binding: OnboardCalculatorFragmentBinding
+
+    companion object {
+        private const val VIEWPAGER_POSITION = "VIEWPAGER_POSITION"
+        fun getInstance(position: Int) = HideBehindCalculator().apply {
+            arguments = bundleOf(VIEWPAGER_POSITION to position)
+        }
+    }
 
     override fun onCreateView(
-        inflater: LayoutInflater, container: ViewGroup?,
-        savedInstanceState: Bundle?
+            inflater: LayoutInflater, container: ViewGroup?,
+            savedInstanceState: Bundle?
     ): View? {
-        val view = inflater.inflate(R.layout.onboard_calculator_fragment, container, false)
+        binding = OnboardCalculatorFragmentBinding.inflate(layoutInflater, container, false)
+        return binding.root
+    }
 
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
         initView(view)
-
-        return view
     }
 
     override fun initView(view: View) {
-        (activity as OnFragmentSelected?)?.hideAppbar()
-
-        view.findViewById<View>(R.id.back_btn).setOnClickListener {
-            activity.onBackPressed()
-        }
-
-        view.findViewById<View>(R.id.calculatorView).setOnClickListener {
-        }
-
-        view.findViewById<View>(R.id.calculatorImg).setOnClickListener {
-        }
-
-        view.findViewById<View>(R.id.calculatorBtn).setOnClickListener {
-            confirmHideBehindCalculator()
+        val position = requireArguments().getInt(VIEWPAGER_POSITION)
+        var drawables = resources.obtainTypedArray(R.array.calculator_skin_array)
+        with(binding) {
+            this.calculatorImg.setImageDrawable(drawables.getDrawable(position))
```

Comment:
why do we need `this`here?

Reply:
here we set the calculator skins images in the viewpager so that the user can choose one skin